jwst/cube_build/src/cube_dq_utils.c

```diff
@@ -308,76 +309,106 @@ int overlap_fov_with_spaxels(int overlap_partial,  int overlap_full,
 //________________________________________________________________________________
 // Routine to setting NIRSpec dq plane for each wavelength plane
 
-int slice_wave_plane_nirspec(int w, int slicevalue,
-		      double roiw_ave,
-		      double *zc,
-		      double *coord1, double *coord2, double *wave,
-		      double *sliceno,
-		      long ncube, int npt,
-		      double *c1_min, double *c2_min, double *c1_max, double *c2_max) {
+long match_wave_plane_nirspec(double wave_plane,
+			      double roiw_ave,
+			      double coord1[],
+			      double coord2[],
+			      double wave[],
+			      double sliceno[], 
+			      long  npt,
+			      double *c1_min, double* c2_min,
+			      double *c1_max, double *c2_max,
+			      int *match_slice){
+
   /* 
      NIRSpec dq plane is set by mapping each slice to IFU wavelength plane 
      This routine maps each slice to sky and finds the min and max coordinates on the sky
      of the slice. 
 
-     w : wavelength plane
+     wave_plane : wavelength of current  plane
      slicevalue : slice # 1 to 30 
      roiw_ave : average roiw for all wavelengths
-     zc : array of wavelengths
-     coord1 : point cloud xi values
-     coord2 : point cloud eta values
+     coord1, coord2: tangent project coordinate of pt cloud
      wave : point cloud wavelength values
-     sliceno : point cloud slice no - starts at 1
-     ncube : number of cube values
+     sliceno: slice value of point cloud.
      npt: number of point cloud elements
 
      return:
-     c1_min, c2_min, c1_max, c2_max
+     c1_min, c2_min, c1_max, c2_max, match_slice
    */
 
-  int ipt, slice, status;
-  double wave_distance, c1, c2;
-  double dvalue = 10000;
-  *c1_min = dvalue;
-  *c2_min = dvalue;
-  *c1_max = -dvalue;
-  *c2_max = -dvalue;
- 
+  long ipt;
+  double wave_distance;
+  double slice;
+  long ii = 0;
+  
+  // initialize the values
+  float minvalue = 10000.0;
+  float maxvalue = -10000.0;
+  for (int i = 0; i< 30; i++){
```

Comment:
Mixed use of whitespace here - some arguments/values have it, others don't. I'm a big fan of consistency. Perhaps add a whitespace before "<".